src/saml/common.ts

```diff
@@ -0,0 +1,99 @@
+import * as querystring from "querystring";
+import { ParsedUrlQuery } from "querystring";
+import * as util from "util";
+import * as zlib from "zlib";
+import * as algorithms from "../algorithms";
+import { keyToPEM } from "../crypto";
+
+const deflateRawAsync = util.promisify(zlib.deflateRaw);
+
+export const getAdditionalParams = (params: {
```

Comment:
Do you think that the signature of this function is now a little more difficult to use since it will no longer determine which additional params are needed based on the operation? Basically, this is now just a glorified object merging utility.

Reply:
> Basically, this is now just a glorified object merging utility

That was also my though. On the other hand, if you look at what is left in _getAdditionalParams, the only thing it does is a ternary on the operationAdditionalParams.

My plan was to eventually get rid of _getAdditionalParams completely (as mentioned in another comment, I initially planed to do it in another PR to let the test untouched for this big code change, but let me know if I should do it here) and instead call directly getAdditionalParams

I have the same thoughts for _requestToUrlAsync. I think I'm not fond of passing this "operation" parameter to select the behavior of the function. 

If we really want to avoid duplicating the logic of selection the right additional params in the function, think I would rather have a function that does just that, and directly use getAdditionalParams.

For example replace
```
this._getAdditionalParams(RelayState, operation, overrideParams)
```
by

```
    return getAdditionalParams({
      relayState,
      globalAdditionalParams: this.options.additionalParams,
      operationAdditionalParams: getAddtionalParamsForOperation(operation)
      overrideParams,
    });
```

With that said, getAdditionalParams is still a glorified object merging utility. I'd say that's fine, because I would find duplicating that code a bit too much, but that could be fine too. Let me know what you think.

Reply:
(I realize that this would makes some things a bit more verbose, but personally I think that verbose is not always a bad thing)

Reply:
I was actually going to comment similarly, verbosity isn't always bad. I really favor obvious code, especially with so many people coming in and out of this code all the time. We want to promote successful and bug-free code writing wherever possible. I personally would be comfortable seeing your suggesting in "by".

Reply:
Should I do that in another PR? This change forces to update the tests which are testing _getAdditionalParams directly, so if we agree on not touching the tests in that PR, it's better to wait the next one.

Reply:
I was thinking that we'd do the entire refactor and then leave a stub for `_getAdditionalParams`.